// Copyright 2023 The MediaPipe Authors.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package mediapipe;

import "mediapipe/framework/calculator.proto";

message TensorToJointsCalculatorOptions {
  extend CalculatorOptions {
    optional TensorToJointsCalculatorOptions ext = 406440177;
  }

  // Number of joints from the output of the model. Calculator will expect the
  // tensor to contain `6 * num_joints + start_index` values.
  optional int32 num_joints = 1;

  // Index to start reading 6 value blocks from.
  optional int32 start_index = 2 [default = 0];
}
